plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -541,7 +545,7 @@ public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, Conn
     {
         DeltaLakeTableHandle table = checkValidTableHandle(tableHandle);
         return getColumns(table.getMetadataEntry()).stream()
-                .collect(toImmutableMap(DeltaLakeColumnHandle::getName, identity()));
+                .collect(toImmutableMap(DeltaLakeColumnHandle::getBaseColumnName, identity()));
```

Comment:
`getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)` can be called for 
`tableHandle` having projected (dereference) columns.

in such case returning column list based on `getColumns(table.getMetadataEntry())` is wrong

currently the code path is not active for Delta (and may never be), so add a check at the top of the method, an no other chhanges needed here